tests/token_token_create_transaction.rs

```diff
@@ -0,0 +1,318 @@
+                                }
+                                ();
+                            }
+                            // Sign with all Privatekeys
+                            None => {
+                                for private_key in private_keys {
+                                    token_create_tx.sign(private_key.clone());
+                                }
+                                ();
+                            }
+                        }
+                    }
+                    None => panic!("Keylist with public keys but no list with private keys to sign the transaction provided."),
+                }
+            ();
+        }
+    }
+
+    // If token supply is finite a maximum supply must be set
+    if used_data.token_supply_type == TokenSupplyType::Finite {
+        match used_data.max_supply {
+            Some(max_supply) => {
+                token_create_tx.max_supply(max_supply);
+                ();
+            }
+            None => panic!("Token Supply Type is finite but no max supply was provided."),
+        }
+    }
+
+    // Execute transaction with max fee of 100 HBar to be sure it gets executed
+    // Get transaction receipt and return only token id
+    let token_id = token_create_tx
+        .max_transaction_fee(Hbar::new(100))
+        .execute(client)
+        .await?
+        .get_receipt(client)
+        .await?
+        .token_id
+        .ok_or_else(|| anyhow::anyhow!("Token creation failed"))?;
+
+    log::info!("successfully created token {token_id}");
+
+    Ok(token_id)
+}
+
+// Fetch token data on chain with TokenInfo query
+async fn get_token_info(client: &Client, token_id: TokenId) -> anyhow::Result<TokenInfo> {
+    let token_info = TokenInfoQuery::new().token_id(token_id).execute(client).await?;
+
+    log::info!("successfully fetched token info of token {token_id}");
+
+    Ok(token_info)
+}
+
+// Compare all data from the token onchain (token_info) with the previously used data (used_data)
+fn check_token_info(token_info: TokenInfo, used_data: &UsedData) {
+    assert_eq!(token_info.name, used_data.name);
+    assert_eq!(token_info.token_type, used_data.token_type);
+    assert_eq!(token_info.symbol, used_data.symbol);
+    assert_eq!(token_info.decimals, used_data.decimals);
+    assert_eq!(token_info.total_supply, used_data.initial_supply);
+    assert_eq!(token_info.treasury_account_id, used_data.account_id);
+    assert_eq!(token_info.default_freeze_status.unwrap(), used_data.default_freeze_status);
+    assert_eq!(token_info.auto_renew_account.unwrap(), used_data.account_id);
+    assert_eq!(token_info.auto_renew_period.unwrap(), used_data.auto_renew_period);
+    assert_eq!(token_info.token_memo, used_data.memo);
+    assert_eq!(token_info.custom_fees, used_data.custom_fees);
+
+    match &used_data.public_key {
+        PublicKeyType::PublicKey(key) => {
+            let key = Key::from(key.clone());
```

Comment:
```suggestion
            let key = Key::Single(*key);
```
(PublicKeys are `Copy`)